docs/layer_reference/recurrent.rst

```diff
@@ -11,13 +11,34 @@ Choice Layer
     :members:
     :undoc-members:
 
-Decision Layer
---------------
+Choice Get Beam Scores Layer
+----------------------------
+
+.. autoclass:: returnn.tf.layers.rec.ChoiceGetBeamScoresLayer
+    :members:
+    :undoc-members:
+
+Choice Get Source Beams Layer
+-----------------------------
+
+.. autoclass:: returnn.tf.layers.rec.ChoiceGetSrcBeamsLayer
+    :members:
+    :undoc-members:
+
+Decide Layer
+------------
 
 .. autoclass:: returnn.tf.layers.rec.DecideLayer
     :members:
     :undoc-members:
 
+Decide Keep Beam Layer
+----------------------
+
+.. autoclass:: returnn.tf.layers.rec.DecideKeepBeamLayer
+    :members:
+    :undoc-members:
```

Comment:
This is really only for internal use. It should not be in the documentation. This should not be used by users.

Reply:
The doc-string sounded like it is intended to be used in a network. This should be fixed then.

Reply:
Well, it can be used (and is also used in some scripts), if you really know what you are doing. If not, you probably don't need it.